Habitica/src/com/habitrpg/android/habitica/ui/fragments/PreferencesFragment.java

```diff
@@ -74,24 +70,23 @@ private void scheduleNotifications() {
 
         String timeval = getPreferenceManager().getSharedPreferences().getString("reminder_time", "19:00");
 
-        if (timeval != null) {
-            String[] pieces = timeval.split(":");
-            int hour = Integer.parseInt(pieces[0]);
-            int minute = Integer.parseInt(pieces[1]);
-            Calendar cal = Calendar.getInstance();
-            cal.set(Calendar.HOUR_OF_DAY, hour);
-            cal.set(Calendar.MINUTE, minute);
-            long trigger_time = cal.getTimeInMillis();
-
-            Intent notificationIntent = new Intent(context, NotificationPublisher.class);
-            notificationIntent.putExtra(NotificationPublisher.NOTIFICATION_ID, 1);
-            notificationIntent.putExtra(NotificationPublisher.NOTIFICATION, getNotification());
-            notificationIntent.putExtra(NotificationPublisher.CHECK_DAILIES, false);
-            PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0, notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-
-            AlarmManager alarmManager = (AlarmManager)context.getSystemService(Context.ALARM_SERVICE);
-            alarmManager.setInexactRepeating(AlarmManager.RTC_WAKEUP, trigger_time, AlarmManager.INTERVAL_DAY, pendingIntent);
-        }
+        String[] pieces = timeval.split(":");
```

Comment:
Ok, sorry. I didn't know that. But shouldn't it always be assigned a value? Because if the sharedPreferences aren't available the second parameter should be assigned...